benchmarks/kubectl-mtb/README.md

```diff
@@ -45,20 +42,99 @@ $ kubectl-mtb get benchmarks
 ### Run the available benchmarks:
 
 ```bash
-$ kubectl-mtb test benchmarks -n "name of tenant-admin namespace" -t "name of tenant service account"
+$ kubectl-mtb test benchmarks -n "name of tenant namespace" --as "name of user/service account"
 ```
 You can mention the profile level of the  benchmark using `-p` flag. 
 
 Example: 
 
 ```bash
-$ kubectl-mtb test benchmarks -n tenant0admin -t system:serviceaccount:tenant0admin:t0-admin0
+$ kubectl-mtb test benchmarks -n tenant0admin --as system:serviceaccount:tenant0admin:t0-admin0
+```
+
+### Create a tenant namespace
+
+Install CRD 
+
+```
+$ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/multi-tenancy/master/tenant/config/crds/tenancy_v1alpha1_tenant.yaml
+$ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/multi-tenancy/master/tenant/config/crds/tenancy_v1alpha1_tenantnamespace.yaml
+```
+
+Install tenant controller 
+
+```bash
+$ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/multi-tenancy/master/tenant/config/manager/all_in_one.yaml
+```
+Create a tenant CR 
+
+```bash
+$ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/multi-tenancy/master/tenant/config/samples/tenancy_v1alpha1_tenant.yaml
+```
+
+Edit the tenant CR and add your user or service account to the tenantAdmins list
+
+```bash
+$ kubectl edit tenant tenant-sample
+```
+
+```yaml
+apiVersion: tenancy.x-k8s.io/v1alpha1
+kind: Tenant
+metadata:
+  labels:
+    controller-tools.k8s.io: "1.0"
+  name: tenant-sample
+spec:
+  # Add fields here
+  tenantAdminNamespaceName: "tenant1admin"
+  tenantAdmins:
+    - kind: User
+      name: your-user
+      namespace: 
+```
+User can do self service namespace creation by creating a tenantnamespace CR in tenant1admin namespace:
+
+Example:
+
+```
+$ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/multi-tenancy/master/tenant/config/samples/tenancy_v1alpha1_tenantnamespace.yaml
+```
+This will create t1-ns1 namespace. If you want to run the benchmarks using this namespace, Make sure that the user can access the namespace using the following command. 
+
+```
+kubectl auth can-i --list --namespace=created-namespace --as your-user
+```
+
+Then, you can run the benchmarks using the following command: 
+
+Example:
+
+```bash
+$ kubectl-mtb test benchmarks -n t1-ns1 --as divya-k8s-access
 ```
 
+### Install Kyverno or Gatekeeper to pass benchmarks
+
+You can use policies to pass the benchmarks. We are currently maintaining [Kyverno](https://github.com/nirmata/kyverno) and [Gatekeeper](https://github.com/open-policy-agent/gatekeeper) policies in this repo which are present [here](https://github.com/kubernetes-sigs/multi-tenancy/tree/master/benchmarks/kubectl-mtb/test/policies)
+
+To install Kyverno, you can run the following command:
+
+```
+kubectl create -f https://github.com/nirmata/kyverno/raw/master/definitions/install.yaml
```

Comment:
Should we also add an command to install the required policies from our repo?

Reply:
Would it be okay, if we point the users to gatekeeper/kyverno README if they want to know how to apply policies? For instance, if we point the users to https://github.com/open-policy-agent/gatekeeper#how-to-use-gatekeeper, they would know the use of constraint templates if they are not familiar and this will help them to apply policy accordingly.

Reply:
@Divya063 - for Kyverno if we have a folder with the required policies they can import as via kubectl as:

```
kubectl create -f https://raw.githubusercontent.com/nirmata/kyverno/master/samples/best_practices/disallow_bind_mounts.yaml
```

Can we provide all policies in a single file?

For Gatekeeper is it possible to do something similar?

We can also reference the web sites for both policy engines for more details.

Reply:
@JimBugwadia In case of gatekeeper, we need to apply both constraint and constraint template (in order to apply a policy), it takes some time for the constraint template to get activated, so I'm not sure if we should have them in the single yaml file. But for kyverno we can do that.

Reply:
Ok, sounds good. 

Lets simplify the process as much as we can for the user and then point to the policy engine repositories for more details.